drools-parser/src/main/java/org/drools/parser/DRLVisitorImpl.java

```diff
@@ -137,7 +137,8 @@ public Object visitLhs(DRLParser.LhsContext ctx) {
     public Object visitLhsPatternBind(DRLParser.LhsPatternBindContext ctx) {
         if (ctx.lhsPattern().size() == 1) {
             Object result = super.visitLhsPatternBind(ctx);
-            PatternDescr patternDescr = (PatternDescr) currentConstructStack.peek().getDescrs().get(0);
+            ConditionalElementDescr parentDescr = currentConstructStack.peek();
+            PatternDescr patternDescr = (PatternDescr) parentDescr.getDescrs().get(parentDescr.getDescrs().size() - 1);
```

Comment:
Just to be sure: in these two lines I see
1) at line 140 we are expecting that `currentConstructStack` is not empty and is last item (the one retrieved by peek) is a `ConditionalElementDescr`
2) at line 141, we are expecting that `parentDescr.getDescrs()` is not empty and that its last element is a  `PatternDescr`
Surely this is what currently happen, but there is no check of whatever kind, and looks a little bit frail, and as soon as one of aforementiond assumptions fails, this methods would throw an uncontrolled exception.
BEside, I have the impression thre are too many if, also nested: maybe it would be worth to split it ?

Reply:
@gitgabrio Thanks. I'll address it in DROOLS-7253

Reply:
👍